### PR TITLE
Show key-events across all live-layout pages (DCR)

### PR DESCRIPTION
**Pull request by OllysCoding**
## What does this change?

For Live/Dead blog pages which are beyond page one (using the `?page:withblock=` syntax), we do not get the key-events data explicitly.

This PR replicates the function of the `KeyEventData` object used in Frontend twirl template, by filtering key events from the body blocks.

Additionally, we've updated the overall DCR key events logic to fetch the most recent summary from the body blocks, which is currently done on Frontend, but was not happening on DCR.

## Does this change need to be reproduced in dotcom-rendering ?

- [x] No
- [ ] Yes (please indicate your plans for DCR Implementation)

## Screenshots

| Before      | After      |
|-------------|------------|
| ![before][] | ![after][] |

[after]: https://user-images.githubusercontent.com/9575458/142891194-4d493b93-1e87-488f-b4df-0d5c4bb9d77c.png
[before]: https://user-images.githubusercontent.com/9575458/142891325-03214dd6-ff7d-4552-9661-39909d9c7bb4.png

(visual bug with key-events spacing should be being addressed separately)

### Tested

- [x] Locally
- [x] On CODE (optional)

### CODE REVIEW

**Review by nitro-marky — APPROVED** [review]
LGTM

**Review by shtukas — APPROVED** [review]

File: common/app/model/dotcomrendering/DotcomRenderingDataModel.scala [hunk 1]
```diff
@@ -228,10 +228,18 @@ object DotcomRenderingDataModel {
...
 
     val bodyBlocks = DotcomRenderingUtils.blocksForLiveblogPage(page, blocks)
-    val keyEvents =
+
+    val allKeyEvents =
       blocks.requestedBodyBlocks
-        .flatMap(blocks => blocks.get("body:key-events"))
-        .getOrElse(Seq.empty[APIBlock])
+        .flatMap(bodyBlocks => bodyBlocks.get("body:key-events"))
+        .getOrElse(blocks.body.fold(Seq.empty[APIBlock])(_.filter(_.attributes.keyEvent.contains(true))))
```
Comment (alinaboghiu) [inline common/app/model/dotcomrendering/DotcomRenderingDataModel.scala:235]: if this is just picking key events from the list of body blocks we can probably avoid using fold, as it's a bit difficult to read
```suggestion
        .getOrElse(blocks.body.filter(_.attributes.keyEvent.contains(true))))
```

Comment (OllysCoding) [reply common/app/model/dotcomrendering/DotcomRenderingDataModel.scala:235]: I think we started by using `.filter`, but we then had to do an additional `.getOrElse(Seq.empty[APIBlock])`, so we changed to a `.fold`, I'm open to either but neither seemed quite right!

Comment (shtukas) [reply common/app/model/dotcomrendering/DotcomRenderingDataModel.scala:235]: I was having a look at tat as well and I find the fold easier to read and understand, but I know those things are personal preferences :)

**Review by alinaboghiu — APPROVED** [review]

**Pull request merged**